exercises/practice/zebra-puzzle/zebra_puzzle.cpp

```diff
@@ -0,0 +1,21 @@
+#include "zebra_puzzle.h"
+
+#include <string>
+
+using namespace std;
+
+namespace zebra_puzzle 
+{
+
+typedef struct 
+{
+    std::string drinksWater;
+    std::string ownsZebra;
+} Solution;
+
+Solution solve() 
+{
+    // IMPLEMENT YOUR LOGIC HERE
+}
```

Comment:
*If* you want to provide a function stub for the students as a hint which function they must implement, and what it returns, declare it in the `.h` file, like in the other exercises that do this.